api/buffer/design.go

```diff
@@ -0,0 +1,542 @@
+	Version("1.0")
+	HTTP(func() {
+		Path("v1")
+		Consumes("application/json")
+		Produces("application/json")
+	})
+	Server("buffer", func() {
+		Host("production", func() {
+			URI("https://buffer.{stack}")
+			Variable("stack", String, "Base URL of the stack", func() {
+				Default("keboola.com")
+				Enum("keboola.com", "eu-central-1.keboola.com", "north-europe.azure.keboola.com")
+			})
+		})
+		Host("localhost", func() {
+			URI("http://localhost:8001")
+		})
+	})
+})
+
+// Service definition
+
+var _ = Service("buffer", func() {
+	Description("A service for continuously importing data to Keboola storage.")
+	// CORS
+	cors.Origin("*", func() {
+		cors.Headers("Content-Type", "X-StorageApi-Token")
+		cors.Methods("GET", "POST", "PATCH", "DELETE")
+	})
+
+	// Set authentication by token to all endpoints without NoSecurity()
+	Security(tokenSecurity)
+	defer AddTokenHeaderToPayloads(tokenSecurity, "storageApiToken", "X-StorageApi-Token")
+
+	// Auxiliary endpoints ---------------------------------------------------------------------------------------------
+
+	Method("ApiRootIndex", func() {
+		Meta("openapi:summary", "Redirect to /v1")
+		Description("Redirect to /v1.")
+		NoSecurity()
+		HTTP(func() {
+			// Redirect / -> /v1
+			GET("//")
+			Meta("openapi:tag:documentation")
+			Redirect("/v1", StatusMovedPermanently)
+		})
+	})
+
+	Method("ApiVersionIndex", func() {
+		Meta("openapi:summary", "List API name and link to documentation.")
+		Description("List API name and link to documentation.")
+		NoSecurity()
+		Result(ServiceDetail)
+		HTTP(func() {
+			GET("")
+			Meta("openapi:tag:documentation")
+			Response(StatusOK)
+		})
+	})
+
+	Method("HealthCheck", func() {
+		NoSecurity()
+		Result(String, func() {
+			Example("OK")
+		})
+		HTTP(func() {
+			GET("//health-check")
+			Meta("openapi:generate", "false")
+			Response(StatusOK, func() {
+				ContentType("text/plain")
+			})
+		})
+	})
+
+	Files("/documentation/openapi.json", "openapi.json", func() {
+		Meta("openapi:summary", "Swagger 2.0 JSON Specification")
+		Meta("openapi:tag:documentation")
+	})
+	Files("/documentation/openapi.yaml", "openapi.yaml", func() {
+		Meta("openapi:summary", "Swagger 2.0 YAML Specification")
+		Meta("openapi:tag:documentation")
+	})
+	Files("/documentation/openapi3.json", "openapi3.json", func() {
+		Meta("openapi:summary", "OpenAPI 3.0 JSON Specification")
+		Meta("openapi:tag:documentation")
+	})
+	Files("/documentation/openapi3.yaml", "openapi3.yaml", func() {
+		Meta("openapi:summary", "OpenAPI 3.0 YAML Specification")
+		Meta("openapi:tag:documentation")
+	})
+	Files("/documentation/{*path}", "swagger-ui", func() {
+		Meta("openapi:generate", "false")
+		Meta("openapi:summary", "Swagger UI")
+		Meta("openapi:tag:documentation")
+	})
+
+	// Main endpoints ---------------------------------------------------------------------------------------------
+
+	// TODO: examples in requests
+
+	Method("CreateReceiver", func() {
```

Comment:
Obecně tady asi ještě chybí examples, jsou tu nějaký jen randomly generated, což není ideální. Tady je zase problém v tom, že pokud mám někde duplikovaný example, tak se ty examples mergnou a vznikne z toho jeden zbytečně dlouhý example. Takže nevím, kam všude můžu dát `Example`, aniž by se to mergovalo s jinými `Example`.

Reply:
Jj, s `examples` je to blbe.
Riesil som to tak, ze som tie examples nastavoval aj nadradenym objektom, ak sa mi tie generovane nepacili.